crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -34,11 +34,21 @@ use std::{
     fmt,
     time::{Duration, Instant},
 };
+use strum::IntoStaticStr;
 use web3::types::{AccessList, TransactionReceipt, U64};
 
 /// Minimal gas price replacement factor
 const GAS_PRICE_BUMP: f64 = 1.125;
 
+/// Error messages which suggest that the node is already aware of the submitted tx thus prompting
+/// us to increase the replacement gas price.
+const ALREADY_KNOWN: &[&str] = &[
+    "Transaction gas price supplied is too low", //openethereum
+    "already known",                             //infura, erigon, eden
+    "INTERNAL_ERROR: existing tx with same hash", //erigon
+    "replacement transaction underpriced",       //eden
```

Comment:
Yes, I changed the `nonce too low` responses back into actual errors (by omitting them from this list) because to me they seemed to be most likely caused by race conditions (i.e. very rare). Those responses are a bit different because technically we shouldn't even continue submitting any more transactions after seeing them but I wasn't sure if that would break the logic.
Would you prefer to have 2 groups of know errors?
1 (same nonce, but underpriced) which results in a gas price bump and 2 (nonce too low) which either gets ignored completely or results in terminating the submission loop early.

Reply:
Ah I see - yes, they are different classifications of errors. I think we should still differentiate between `nonce too low` and other errors, since if they appear in a "transaction re-send" it is benign and just means that the transaction got mined, but we didn't see it yet.

Reply:
Added the second category of error messages.